### PR TITLE
Bigtest website migration to monorepo

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation
We're creating a new website for `bigtest`. I created the landing page on my own repository and it's now being moved here hence the sizable pull request.

## Approach
The website was constructed using `gatsby`, `netlify`, and `styled-components`.

Group

- Added `packages/website` as ignore argument to publish preview workflow.

## Future TODOs
- Configure Netlify from its website
- Setup Mailchimp account for newsletter sign up
- I included a list of todos for the website in [`packages/website/todos/notes.md`](https://github.com/bigtestjs/bigtest/blob/mk/website/packages/website/todos/notes.md)

## Learning
- [Current preview of this commit from my repository](https://mk-bigtestjs-io.netlify.com/)

### CODE REVIEW

**Comment by minkimcello** [comment]
The preview action is failing because package is marked private. Can't make it not-private it seems because yarn workspace requires it to be private?

I know we added a feature in the preview action in `resideo/actions` to skip private packages but I'll have a look to see how their monorepo is different and why it allows them to have public packages.

**Review by cowboyd — COMMENTED** [review]
This is awesome @minkimcello, thank you!

One thing to note on the "sizable" pull request. Step 1 should pretty much always be "deploy to production". This means that the netlify step in the TODO section should actually come first. That way, you have a reliable pipeline to gradually deliver the website.

Remember, whether its a new app, a new website, or a new service, we should always be thinking about getting it into production first, and only _then_ iterating on it.

As for the package publishing. I think it's safe to have auto publishing explicitly ignore the website since it never needs to go to NPM. This will use netlify previews.

**Comment by jorgelainfiesta** [comment]
BigTest's new code is being written in Typescript, should we evaluate whether we should do so on this website too?

**Comment by jorgelainfiesta** [comment]
I noticed that you're using CSS vars in the code. However, we want to avoid having any global CSS. Instead, we want explicit dependencies. For this reason, we should use variables from a theme. This will also allow us to change themes programatically (ie. for dark theme).

You can read more about Themes in styled-components here: https://styled-components.com/docs/advanced#theming

File: packages/website/src/components/row.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  align-items: center;
+`;
+
+const RowComponent = styled.div`
+  margin-left: var(--space-half-vw);
+`;
+
+const Row = ({ className, children }) => {
+  return (
+    <RowContainer className={css(className)}>
```
Comment (jorgelainfiesta) [inline packages/website/src/components/row.js]: We should not use `className` anywhere, instead we want to work only with the `styled` API

File: packages/website/src/styles/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import styled from 'styled-components';
+
+const logo_height = "calc(var(--size-base)*2/1.75)";
+
+const Index = styled.div`
+  h1 {
```
Comment (jorgelainfiesta) [inline packages/website/src/styles/index.js]: This is not an adequate way to declaring styles. We want to favor `styled-components` for each tag and avoid leaking styles around.

File: packages/website/src/components/row.js (hunk 1 above)
Comment (minkimcello) [reply packages/website/src/components/row.js]: okay I'll fix it

File: packages/website/src/styles/index.js (hunk 2 above)
Comment (minkimcello) [reply packages/website/src/styles/index.js]: And this too

**Comment by minkimcello** [comment]
@taras Does Bigtest org have its own Netlify account? Can we point its site to this repository to see if it deploys preview properly?

**Comment by taras** [comment]
@minkimcello we might want to move this to Frontside before we setup netlify. It gets confusing.

**Comment by cowboyd** [comment]
> @minkimcello we might want to move this to Frontside before we setup netlify. It gets confusing.

Yes, let's do that.

**Comment by minkimcello** [comment]
@jorgelainfiesta Should I get a 👍 before merge?

**Review by jorgelainfiesta — APPROVED** [review]
We'll work on refactoring the styles in different PRs. For now, let's merge it to reduce complexity in the upcoming changes.

**Pull request merged**